### ISSUE TITLE
AstropyDeprecationWarning from write_map / new_table

### ISSUE THREAD

**Issue by cdeil** [healpy/healpy#253]
With the latest Astropy I'm getting this deprecation warning:
```python
>>> import numpy as np
>>> data = np.zeros(12, dtype=float)
>>> import healpy as hp
>>> hp.write_map('/tmp/data.fits', data)
WARNING: AstropyDeprecationWarning: The new_table function is deprecated and may be removed in a future version.
        Use :meth:`BinTableHDU.from_columns` for new BINARY tables or :meth:`TableHDU.from_columns` for new ASCII tables instead. [healpy.fitsfunc]
WARNING: AstropyDeprecationWarning: The use of header.update() to add new keywords to a header is deprecated.  Instead, use either header.set() or simply `header[keyword] = value` or `header[keyword] = (value, comment)`.  header.set() is only necessary to use if you also want to use the before/after keyword arguments. [astropy.io.fits.header]
```

See https://github.com/healpy/healpy/search?utf8=%E2%9C%93&q=new_table

**Comment by zonca** [comment]
This should be a pretty easy fix, just rewrite the fits writing without using `new_table` but instead using the other `astropy` FITS building functions.
Hope somebody can help and contribute this fix with a Pull Request.

**Comment by lpsinger** [comment]
Unfortunately, we cannot install Astropy on the Scientific Linux 6 LIGO computing clusters; we are using Pyfits 2.3.1 because we do not have a new enough version of Numpy to install Astropy. I am checking when an update to Scientific Linux 7 is anticipated; this update would bring Numpy to 1.7.1 and would permit us to use Astropy instead of the very old version of Pyfits.

**Comment by astrofrog** [comment]
@lpsinger as I mentioned in a separate issue, I wonder whether it would be worth splitting off the development to have a long term supported release that would be suitable for the LIGO cluster (if the update to SL7 is not happening soon) while being able to go ahead with making changes like these in the master branch?

**Comment by lpsinger** [comment]
> @lpsinger as I mentioned in a separate issue, I wonder whether it would be worth splitting off the development to have a long term supported release that would be suitable for the LIGO cluster (if the update to SL7 is not happening soon) while being able to go ahead with making changes like these in the master branch?

Tempting. Dealing with obsolete versions of Numpy is a huge pain. The Healpy build, however, is unusual enough already; I'd favor sticking with the current single development line model, though of course this is up to @zonca.

I have further information from our admins: we will be using SL6 for at least the first Advanced LIGO observing run (O1), and re-evaluate updating to SL7 after that. However, we may be able to upgrade from SL6.1 to SL6.5. This would get us from Numpy 1.3.0 to Numpy 1.4.1 and enable us to use Astropy 0.2.5. That should be enough to modernize the FITS support and clear out the deprecated calls, right?

**Comment by astrofrog** [comment]
@lpsinger - I think that would be enough to fix this issue (astropy 0.2.5) but note that this won't help for https://github.com/healpy/healpy/issues/247 which will require 0.4+. An alternative to having the dual-branch development model would be to have wrappers for the FITS functionality and then in those branch differently depending on whether one is using Astropy or PyFITS?

**Comment by lpsinger** [comment]
@astrofrog: It looks like both Astropy 0.2.5 and 0.4+ have galactic <-> equatorial coordinate conversion, but with slightly different APIs. We could switch between the two with a `try`/`except` block or by switching on the Astropy version. We can do without support for ecliptic coordinates on the LIGO computing clusters.

**Comment by astrofrog** [comment]
@lpsinger - ah yes, you were saying that the coordinate stuff wasn't really needed at all for LIGO anyway? If that is the case, then I'd suggest not even bothering supporting astropy < 0.4. That is, if astropy is installed, it should be >= 0.4.

**Comment by lpsinger** [comment]
Ah, but we'd like to get rid of obsolete Pyfits calls too, which Astropy >= 0.2.5 would take care of.

**Comment by zonca** [comment]
@lpsinger @astrofrog I also think it would be too much effort to maintain two branches.

@cdeil would you like to contribute a fix that replaces `new_table` with standard `pyfits` code?

**Comment by lpsinger** [comment]
Fixed in #346.
